mParticle-Leanplum/MPKitLeanplum.m

```diff
@@ -158,6 +158,72 @@ - (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MP
     return execStatus;
 }
 
+#pragma mark e-Commerce
+- (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
+    
+    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:@(MPKitInstanceLeanplum) returnCode:MPKitReturnCodeSuccess forwardCount:0];
+    
+    if (commerceEvent.type == MPEventTypePurchase) {
+        NSMutableDictionary *baseProductAttributes = [[NSMutableDictionary alloc] init];
+        NSDictionary *transactionAttributes = [commerceEvent.transactionAttributes beautifiedDictionaryRepresentation];
+        
+        if (transactionAttributes) {
+            [baseProductAttributes addEntriesFromDictionary:transactionAttributes];
+        }
+        
+        NSDictionary *commerceEventAttributes = [commerceEvent beautifiedAttributes];
+        NSArray *keys = @[kMPExpCECheckoutOptions, kMPExpCECheckoutStep, kMPExpCEProductListName, kMPExpCEProductListSource];
+        
+        for (NSString *key in keys) {
+            if (commerceEventAttributes[key]) {
+                baseProductAttributes[key] = commerceEventAttributes[key];
+            }
+        }
+        
+        NSArray *products = commerceEvent.products;
+        NSString *currency = commerceEvent.currency ? : @"USD";
+        NSMutableDictionary *properties;
+        
+        for (MPProduct *product in products) {
+            // Add relevant attributes from the commerce event
+            properties = [[NSMutableDictionary alloc] init];
+            if (baseProductAttributes.count > 0) {
+                [properties addEntriesFromDictionary:baseProductAttributes];
+            }
+            
+            // Add attributes from the product itself
+            NSDictionary *productDictionary = [product beautifiedDictionaryRepresentation];
+            if (productDictionary) {
+                [properties addEntriesFromDictionary:productDictionary];
+            }
+            
+            // Strips key/values already being passed, plus key/values initialized to default values
+            keys = @[kMPProductAffiliation, kMPExpProductCategory, kMPExpProductName];
```

Comment:
This list is already stripped down from the one in appboy. I assumed product affiliation and category were set to default values so that's why I left them.